pkg/defaultcomponents/defaults_test.go

```diff
@@ -104,46 +103,8 @@ func TestComponents(t *testing.T) {
 	assert.NotNil(t, processors["tail_sampling"])
 	assert.NotNil(t, processors["k8sattributes"])
 
-	// Ensure that the components behind feature gates aren't included
-	assert.Nil(t, receivers["filelog"])
-	assert.Nil(t, exporters["awscloudwatchlogs"])
-	assert.Nil(t, extensions["file_storage"])
-}
-
-func TestEnabledFeatureGate(t *testing.T) {
```

Comment:
You should adjust this test case to verify that the components are not included if the feature gate is disabled. Basically the inverse of this current test.